File: 06_상속/src/Ex06_04_상속.java

class Product {
	private String code;
	private String company;
	
	Product(String code, String company) {
		this.code = code;
		this.company = company;
	}
	
	void show() {
		System.out.println("code : " + code);
		System.out.println("company : " + company);
	}
	
	String getCode() {
		return this.code;
	}
	
	String getCompany() {
		return this.company;
	}
}

class Computer extends Product {
	
	private String os;
	private int ram;
	
	Computer(String code, String company, String os, int ram) {
		super(code, company);
		this.os = os;
		this.ram = ram;
	}
	
	@Override
	void show() {
		super.show();
		System.out.println("os : " + os);
		System.out.println("ram : " + ram);
	}
}

class Book extends Product {
	
	private String title;
	private String author;
	
	Book(String code, String company, String title, String author) {
		super(code, company);
		this.title = title;
		this.author = author;
	}
	
	String getTitle() {
		return this.title;
	}
	
	String getAuthor() {
		return this.author;
	}
}

public class Ex06_04_상속 {
	public static void main(String[] args) {

		Product p = new Product("P01", "제조사");
		p.show();
		
		System.out.println();
		
		Computer c = new Computer("Co2", "삼성", "window10", 64);
		c.show();
		
		System.out.println();
		
		Book b = new Book("B03", "중앙출판사", "자바", "이무진");
		
		System.out.println("code : " + b.getCode());
		System.out.println("company : " + b.getCompany());
		System.out.println("title : " + b.getTitle());
		System.out.println("author : " + b.getAuthor());
	
		System.out.println("===================================");
		
		Product[] pd = {
				new Product("P01", "제조사"),
				new Computer("Co2", "삼성", "window10", 64),
				new Book("B03", "중앙출판사", "자바", "이무진")
		};
		
		for(Product p2 : pd) {
			p2.show();
			System.out.println();
		}
	}
}

File: PythonP/06_상속/Ex06_04_상속.py
'''
Created on 2022. 5. 26.

@author: admin
'''

class Product:
    def __init__(self, code, company):
        self.__code = code
        self.__company = company
        
    def show(self):
        print("code : {}".format(self.__code))
        print("company : {}".format(self.__company))
        
    @property
    def code(self):
        return self.__code
    
    @property
    def company(self):
        return self.__company
        
class Computer(Product):
    def __init__(self, code, company, os, ram):
        Product.__init__(self, code, company)
        self.__os = os
        self.__ram = ram
        
    def show(self):
        Product.show(self)
        print("os : {}".format(self.__os))
        print("ram : {}".format(self.__ram))
        
class Book(Product):
    def __init__(self, code, company, title, author):
        super().__init__(code, company)
        self.__title = title
        self.__author = author
    
    @property
    def title(self):
        return self.__title
    
    @property
    def author(self):
        return self.__author
    

p = Product("P01", "제조사")
p.show()

print()

c = Computer("C02", "삼성", "window10", 64)
c.show()

print()

b = Book("B03", "중앙출판사", "자바", "이무진")
print("code : {}".format(b.code))
print("company : {}".format(b.company))
print("title : {}".format(b.title))
print("author : {}".format(b.author))